File: vbnet/App_Code/Modules/jquery/JQueryModule.vb

Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc

Public Class JQueryExternalModule
    Implements IExternalModule
    Implements IModule
    Private _ctrl As Controller
    Public Sub Load(model As Object, ParamArray args As Object()) Implements IExternalModule.Load
        model.JQuery = Me
        If args IsNot Nothing Then
            _ctrl = args.OfType(Of Controller)().FirstOrDefault()
        End If
    End Sub

    Public Function GetViewOfType(type As String) As String Implements IModule.GetViewOfType
        If type = "corejs" Then
            Return String.Format("<script src='{0}'></script>", _ctrl.Url.Content("/Scripts/jquery-2.1.4.js"))
        End If

        Return Nothing
    End Function
End Class

File: csharp/App_Code/Modules/jquery/JQueryModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

public class JQueryExternalModule : IExternalModule, IModule
{
    private Controller _ctrl;
    public void Load(dynamic model, params object[] args)
    {
        model.JQuery = this;
        if (args != null)
            _ctrl = args.OfType<Controller>().FirstOrDefault();
    }

    //public string Name
    //{
    //    get { return string.Empty; }
    //}

    //public string View
    //{
    //    get {return "raw"; }
    //}

    //public object Model
    //{
    //    get { return @"<script src='/Scripts/jquery-2.1.4.js'></script>"; }
    //}

    public string GetViewOfType(string type)
    {
        if (type == "corejs")
            return string.Format("<script src='{0}'></script>" ,_ctrl.Url.Content("/Scripts/jquery-2.1.4.js"));

        return null;
    }
}